pennylane/devices/default_qubit.py

```diff
@@ -630,18 +630,12 @@ def _apply_state_vector(self, state, device_wires):
         state = self._asarray(state, dtype=self.C_DTYPE)
         n_state_vector = state.shape[0]
 
-        if state.ndim != 1 or n_state_vector != 2 ** len(device_wires):
+        if len(qml.math.shape(state)) != 1 or n_state_vector != 2 ** len(device_wires):
```

Comment:
This change was not necessary for avoiding problem with jit, right?

Reply:
It was, but only for TensorFlow, not JAX 🙂 

Without JIT:

```python
>>> x = tf.constant([0.2, 0.1])
>>> cost = lambda x: x.ndim
>>> cost(x)
1
```

With JIT:

```python
>>> tf.function(cost)(x)
 AttributeError: 'Tensor' object has no attribute 'ndim'
```

It seems that TF abstract tensors no longer have the `ndim` attribute.